### PR TITLE
feat: crud task according to workflow

### PR DESCRIPTION
**Pull request by jean-michelet**
Issue #15 

I created a task CRUD with workflow configuration.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,11 +8,12 @@
     "test": "test"
   },
   "scripts": {
-    "build": "rm -rf dist && tsc",
-    "watch": "npm run build -- --watch",
+    "start": "npm run build && fastify start -l info dist/app.js",
+    "build": "tsc",
+    "watch": "tsc -w",
+    "dev": "npm run build && concurrently -k -p \"[{name}]\" -n \"TypeScript,App\" -c \"yellow.bold,cyan.bold\" \"npm:watch\" \"npm:dev:start\"",
+    "dev:start": "fastify start --ignore-watch=.ts$ -w -l info -P dist/app.js",
```
Comment (jean-michelet) [inline package.json:15]: Improve the dev mode experience. Inspired by: https://github.com/fastify/fastify-cli/blob/master/generate.js#L54

File: src/app.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ export default async function serviceApp(
   fastify: FastifyInstance,
   opts: FastifyPluginOptions
 ) {
+  delete opts.skipOverride // This option only serves testing purpose
```
Comment (jean-michelet) [inline src/app.ts:13]: As I feared, I think we've introduced a problem here: https://github.com/fastify/fastify-cli/pull/742/files#r1689541348

For example, `@fastify/mysql` [passes all the given options directly to `mysql2`](https://github.com/fastify/fastify-mysql/blob/10a2a8e34911d1982ad3556b03424c276e7ccdcf/index.js#L62), which triggers warnings from `mysql2`:
```
Ignoring invalid configuration option passed to Connection: skipOverride. This is currently a warning, but in future versions of MySQL2, an error
will be thrown if you pass an invalid configuration option to a Connection
```

Maybe it's also a bad design to propagate all the options in the autoloader? Like the cli generated template suggest to do:
```ts
export default async function serviceApp(
  fastify: FastifyInstance,
  opts: FastifyPluginOptions
) {
  await fastify.register(fastifyAutoload, {
      dir: path.join(import.meta.dirname, "plugins"),
      options: { ...opts } // opts propagated here
  });
  
  // code...
}
```

File: src/plugins/custom/workflow.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import fp from 'fastify-plugin';
+import { MultiOriginTransition, Transition, Workflow } from '@jean-michelet/workflow';
```
Comment (jean-michelet) [inline src/plugins/custom/workflow.ts:2]: This is a component I created inspired by [symfony/workflow](https://symfony.com/doc/current/components/workflow.html). You can read the documentation here, I am willing to create an official Fastify plugin if you are interested.

File: src/schemas/tasks.ts [hunk 4]
```diff
@@ -1,6 +1,55 @@
...
+} as const;
+
+export const TaskStatus = {
+  New: 'new',
+  InProgress: 'in-progress',
+  OnHold: 'on-hold',
+  Completed: 'completed',
+  Canceled: 'canceled',
+  Archived: 'archived'
+} as const;
```
Comment (jean-michelet) [inline src/schemas/tasks.ts:19]: A technic to create safer enums.

**Comment by jean-michelet** [comment]
@melroy89 :wave:

File: src/app.ts (hunk 2 above)
Comment (melroy89) [reply src/app.ts:13]: Not sure what to say here, but you are right. Passing all default options that results into this error, is NOT good.

**Comment by melroy89** [comment]
Sorry about my ignorance, but why do we need all those workflow code in a demo app? The CRUD code all seems fine to me. But what does all that workflow code do? Which has even an dependency outside this demo app with `@jean-michelet/workflow`.

Why not just keep the demo clean, and focus on the CRUD part only? Or do I miss something.

**Comment by jean-michelet** [comment]
@gurgunday @climba03003 
Any feedback?

File: src/plugins/custom/workflow.ts (hunk 3 above)
Comment (melroy89) [reply src/plugins/custom/workflow.ts:2]: I still thing, this shouldn't be part of the Demo repo.

**Review by climba03003 — COMMENTED** [review]
IMO, the demo should follow some existing well defined schema.
For example https://petstore.swagger.io/
It shows `authorization`, `linkage between entity`, `multipart handling`, etc.

**Comment by jean-michelet** [comment]
I can take inspiration from it, but maybe we can stick to a task application. I can complete this PR with users endpoints, I just wanted a feedback on a pure CRUD. Autorisation is for a next step.

**Comment by jean-michelet** [comment]
TODO:
- Add users endpoint

**Review by mcollina — APPROVED** [review]
lgtm

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by jean-michelet** [comment]
Can do Users on an other PR.

File: README.md [hunk 5]
```diff
@@ -41,11 +41,6 @@ To build the project:
 npm run build
 ```
 
-In dev mode, you can use:
-```bash
-npm run watch
```
Comment (melroy89) [inline README.md:46]: is watch not working or?

Comment (jean-michelet) [reply README.md:46]: It does work, but will people use it now that we run it concurrently with the dev server?

File: src/app.ts [hunk 6]
```diff
@@ -6,10 +6,21 @@ import path from "node:path";
 import fastifyAutoload from "@fastify/autoload";
 import { FastifyInstance, FastifyPluginOptions } from "fastify";
 
+export const options = {
+  ajv: {
+    customOptions: {
+      coerceTypes: "array", 
+      removeAdditional: "all",
+      allErrors: true
```
Comment (climba03003) [inline src/app.ts:14]: I don't think we should enable `allErrors`.

**Review by climba03003 — APPROVED** [review]
I still think that we should follow some existing example.
But since you are the companion of this repository, your motivation on continuing the work is more important.

The only concern is about the `ajv.allErrors`, others code LGTM.

**Review by gurgunday — APPROVED** [review]
lgtm

**Pull request merged**